Teach the sidebar to hide unavailable functionality by default

algernon opened the pull request:
Recent firmware supports a `plugins` command, letting us query the list of available plugins. We can leverage this information to hide functionality from the sidebar for which firmware-side support does not exist - at least optionally. This pull request does just that.

The parts of the Sidebar which depend on firmware-side plugins will now check if they should make those functions visible. They do so via the new `usePluginVisibility()` hook that checks the available plugin list, and also consults the UI settings, which can override the default hiding behaviour.

The result is a considerably cleaner sidebar, with no fluff. With hiding functionality not supported by the firmware, we avoid surprises: a very frequently asked question on Discord is why "Secondary actions" don't work, and the answer is always: because the plugin is not enabled. With this PR, we simply won't show that section if `Qukeys` isn't enabled in the firmware.

For a very bare-bones firmware with only `EEPROMKeymap` and `Focus` enabled, the sidebar looks like this:

![Screenshot from 2022-05-27 11-42-59](https://user-images.githubusercontent.com/17243/170675871-58c4172c-ad25-4b9d-9102-7cba83ce6768.png)

We can, of course, disable this feature, and show all possible keys:

![Screenshot from 2022-05-27 11-42-49](https://user-images.githubusercontent.com/17243/170675971-66f96c9c-f65f-4cb2-92e0-5645ab830cd6.png)

If the `plugins` command is not supported by the firmware, we fall back to showing everything.

algernon commented:
I'm not entirely happy with the label of the setting in Preferences. Ideally, there'd be a bit more explanation. I plan to have another pass at the Preferences screen in the next few days, adding more help text and making it look even nicer, so worst case, we'd update the label at that time.

obra commented on src/renderer/screens/Editor/Sidebar/LeaderKeys.js:
```diff
@@ -15,13 +15,17 @@
  */
 
-import React from "react";
+import React, { useContext, useEffect, useState } from "react";
```
Do we really need these extra imports?

algernon replied:
We do not, that's a leftover from my first iteration.

obra commented on src/renderer/screens/Preferences/UserInterface.js:
```diff
@@ -49,6 +50,11 @@ function UserInterfacePreferences(props) {
   };
 
+  const toggleAllowAllKeys = () => {
+    settings.set("ui.allowAllKeys", !allowAllKeys);
```
Let's use a setting name that's more descriptive? 

Maybe ui.hideFeaturesNotAvailableInCurrentFirmware? 

It's wordy but also, I think, a lot easier to understand

obra commented:
The core feature is <3

For the phrasing on the settings screen, with a default to ON:

[off | ON] "Hide features not supported by your keyboard's current firmware"

The lighter colored text underneath can read:

"When enabled, Chrysalis will hide configuration options for features that your keyboard's firmware doesn't support."

algernon commented:
Removed the unnecessary includes, updated the preference name, and the texts too, and force pushed the result.

Preferences looks like this:

![Screenshot from 2022-05-27 20-59-36](https://user-images.githubusercontent.com/17243/170774101-bd0e18c2-a07f-4590-a20f-8b2836a2fcc9.png)
